### PR TITLE
team 1's placeable features to main!!

### PR DESCRIPTION
**Pull request by cmp6**
core feature ticket: [Quality of life and some place-able items](#292)

Adds 3 types of placeable items:
- Sprinklers & pumps (#310)
    - sprinklers look like pipes and water the ground around them, but the need a pump to power them.
    - 1 pump can power every sprinkler if they're all connected!
- Fences & gates (#303)
    - fences block entities movements and gates can be opened and close via 'E', the interact key. 
- Chests (#313)
    - are just for show at the moment 😢, but will contain their own sort of inventory as you'd expect.
    - also interactable via the 'E' key.

Notes:
- All these placeables can be spawned in using the >addItem console command, e.g:
    >addItem sprinkler
- All these placeables can be removed using the shovel.
- Sprinklers watering amount, fequency, and AOE can all be easily modified, just @ me. 
- Chests are waiting on the inventory re-work to be finished.

### CODE REVIEW

File: source/core/assets/images/placeable/fences/gate.png
Comment (JTrenerry) [inline source/core/assets/images/placeable/fences/gate.png:1]: Can we remove this or something

File: source/core/src/main/com/csse3200/game/components/placeables/SprinklerComponent.java [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+            new Vector2(x + 1, y),
+            new Vector2(x + 2, y),
+            new Vector2(x - 1, y),
+            new Vector2(x - 2, y),
+            // top right, bottom right, top left, bottom left.
+            /*
+            new Vector2(x + 1, y + 1),
+            new Vector2(x + 1, y - 1),
+            new Vector2(x - 1, y + 1),
+            new Vector2(x - 1, y - 1)
```
Comment (JTrenerry) [inline source/core/src/main/com/csse3200/game/components/placeables/SprinklerComponent.java:276]: to remove?

File: source/core/src/main/com/csse3200/game/entities/factories/PlaceableFactory.java [hunk 2]
```diff
@@ -4,73 +4,113 @@
...
+     * Creates a pump Entity - used in conjunction with sprinklers
+     * @return the pump that was made
+     */
+    public static Entity createPump() {
+        EntityType type = EntityType.Pump;
+        type.setPlaceableCategory(PlaceableCategory.Sprinklers);
+        Entity pump = createBasePlaceable(type);
+        // set temp texture to differentiate from other entities
+        pump.addComponent(new DynamicTextureRenderComponent("images/placeable/sprinkler/pump.png"));
+        //pump.getComponent(DynamicTextureRenderComponent.class).setTexture("images/placeable/sprinkler/pump.png");
```
Comment (JTrenerry) [inline source/core/src/main/com/csse3200/game/entities/factories/PlaceableFactory.java:95]: to remove?

**Review by JTrenerry — APPROVED** [review]
Can we please clean up what I commented on (very small just remove things) then should be good. Also I am 100% we have gate sprites so maybe get Peter to commit to make them in.

File: source/core/assets/images/placeable/fences/gate.png
Comment (86LAK) [inline source/core/assets/images/placeable/fences/gate.png:1]: why is box boy back

File: source/core/assets/images/placeable/fences/gate_open.png
Comment (86LAK) [inline source/core/assets/images/placeable/fences/gate_open.png:1]: same here for ghost

**Review by p-lahanas — APPROVED** [review]

**Review by 86LAK — APPROVED** [review]
lgtm

**Pull request merged**